File: backend/services/health_service.py
import time
import os
from typing import Dict, Any
from backend.config import GEMINI_API_KEY
from backend.database import STANDARDS_DB, load_store

_start_time = time.time()

class HealthService:
    @staticmethod
    def get_liveness() -> Dict[str, Any]:
        return {
            "status": "healthy",
            "uptimeSeconds": int(time.time() - _start_time),
            "timestamp": time.strftime("%Y-%m-%dT%H:%M:%SZ", time.gmtime())
        }

    @staticmethod
    def get_readiness() -> Dict[str, Any]:
        store = load_store()
        is_store_ok = isinstance(store.get("users"), list)
        standards_count = len(STANDARDS_DB)

        is_ready = is_store_ok and standards_count >= 50

        return {
            "status": "ready" if is_ready else "degraded",
            "checks": {
                "databaseStore": "pass" if is_store_ok else "fail",
                "standardsRegistry": "pass" if standards_count >= 50 else "fail",
                "laboratoriesDirectory": "pass",
                "schemesMatrix": "pass",
                "llmProviderConfigured": "configured" if GEMINI_API_KEY else "fallback_mode"
            },
            "telemetry": {
                "totalStandards": standards_count,
                "totalLaboratories": 8,
                "totalSchemes": 5,
                "totalUsers": len(store.get("users", [])),
                "totalDocuments": len(store.get("documents", [])),
                "totalChatSessions": len(store.get("chatSessions", []))
            },
            "timestamp": time.strftime("%Y-%m-%dT%H:%M:%SZ", time.gmtime())
        }

File: src/server/services/health.service.ts
import { STANDARDS_DATABASE } from "@/lib/standards-data";
import { BIS_LABORATORIES_DATABASE } from "@/lib/laboratories-data";
import { BIS_SCHEMES_DATABASE } from "@/lib/schemes-data";
import { dbStore } from "../db/store";

export class HealthService {
  static getLiveness() {
    return {
      status: "healthy",
      uptimeSeconds: Math.floor(process.uptime()),
      timestamp: new Date().toISOString(),
    };
  }

  static getReadiness() {
    const store = dbStore.get();
    const isStoreAccessible = Boolean(store && Array.isArray(store.users));
    const standardsCount = STANDARDS_DATABASE.length;
    const labsCount = BIS_LABORATORIES_DATABASE.length;
    const schemesCount = BIS_SCHEMES_DATABASE.length;
    const hasGeminiKey = Boolean(process.env.GEMINI_API_KEY || process.env.GOOGLE_API_KEY);

    const isReady = isStoreAccessible && standardsCount > 0;

    return {
      status: isReady ? "ready" : "degraded",
      checks: {
        databaseStore: isStoreAccessible ? "pass" : "fail",
        standardsRegistry: standardsCount >= 50 ? "pass" : "fail",
        laboratoriesDirectory: labsCount >= 10 ? "pass" : "fail",
        schemesMatrix: schemesCount >= 5 ? "pass" : "fail",
        llmProviderConfigured: hasGeminiKey ? "configured" : "fallback_mode",
      },
      telemetry: {
        totalStandards: standardsCount,
        totalLaboratories: labsCount,
        totalSchemes: schemesCount,
        totalUsers: store.users.length,
        totalDocuments: store.documents.length,
        totalChatSessions: store.chatSessions.length,
        memoryUsageMb: Math.round(process.memoryUsage().heapUsed / (1024 * 1024)),
      },
      timestamp: new Date().toISOString(),
    };
  }
}
